view_conversations/conv-viewer-webapp/src/app/app.component.ts

```diff
@@ -142,4 +218,38 @@ export class AppComponent implements OnInit {
                   delete this.inFlightRequest;
                 });
   }
+
+  submitBrowse() {
+    console.log('model-based browse form submitted');
+    console.log(this.browseForm.value);
+    this.errorMessage = null;
+    this.updateLocationHash();
+
+    this.inFlightBrowseRequest =
```

Comment:
At the moment, there is no check for if an existing request is in flight... 

I think the pattern you want for this is:
 * some kind of UI (use this one: https://material.angular.io/components/progress-bar/overview or the spinner) get started for while things are loaded. 
 * If requests can overlap each other: cancel the old one. If they can't; cancel the new one until the old one has failed (or cancel the old and send a new one)
 * Once loading has completed end the spinner. 
 * Don't allow multiple callbacks unless you really want them. 
 * See if you can pull out the callback logic into a separate function that you can test instead of being an anonymous function you can't test.

Reply:
inFlightBrowseRequest is a feature in the view-conversation before I started editing. Should we just change this into a different tool or we re-consider why this was used at the first place?